Filtered_kernel/include/CGAL/internal/Static_filters/Collinear_3.h

```diff
@@ -77,32 +77,18 @@ class Collinear_3
         upper_bound_1 = max1;
 
       int int_tmp_result;
-      if (lower_bound_1 < 5.00368081960964635413e-147)
+
+      if ( (lower_bound_1 < 5.00368081960964635413e-147) || (upper_bound_1 > 1.67597599124282407923e+153) )
         return Base::operator()(p, q, r);
```

Comment:
IIUC, if all the points have the same x coordinate, we end up here, it seems a bit too early to go to the exact computation, we could check the other projections. Maybe that falls under "further suggested improvements".

Reply:
I agree with Marc: the static filter should only end up on the exact computation if all the possible filtering have failed. The *raison d'être* of those static filters is to quickly find a reason to tell that the points are not collinear.

Points `p`, `q` and `r` and collinear if and only if the two vector `pq` and `pr` are collinear, that is if and only if their cross-product is null. The cross-product has three coordinates that are each 2×2 determinants. That gives exactly three opportunities for the static filters to prove that the three points are not collinear. The `return Base::operator()(p, q, r);` should be only at the end, when all three opportunities have failed.

Reply:
> Maybe that falls under "further suggested improvements".

I meant what you suggest for `collinearC3()`.   Also I am wondering if  we could not profit from a function `is_determinant_zero()`  which would save us a subtraction.

Reply:
>  Also I am wondering if we could not profit from a function `is_determinant_zero()` which would save us a subtraction.

We have sign_of_determinant, although compilers may not be perfect at simplifying sign(double)==0.

Reply:
In fact I was wondering about that for rational types.

Reply:
Indeed, and for rationals, cmp can be quite a bit more expensive than ==.